src/Microsoft.DotNet.Wpf/src/PresentationFramework/Microsoft/Win32/CommonItemDialog.cs

```diff
@@ -0,0 +1,693 @@
+//              variants. It provides a common options storage and events handling.
+//
+
+namespace Microsoft.Win32
+{
+    using MS.Internal;
+    using MS.Internal.AppModel;
+    using MS.Internal.Interop;
+    using MS.Win32;
+
+    using System;
+    using System.ComponentModel;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Runtime.InteropServices;
+    using System.Text;
+    using System.Windows;
+
+    using HRESULT = MS.Internal.Interop.HRESULT;
+
+    /// <summary>
+    ///    Provides a common base class for wrappers around both the
+    ///    File Open and File Save common dialog boxes.  Derives from
+    ///    CommonDialog.
+    ///
+    ///    This class is not intended to be derived from except by
+    ///    the OpenFileDialog and SaveFileDialog classes.
+    /// </summary>
+    public abstract class CommonItemDialog : CommonDialog
+    {
+        //---------------------------------------------------
+        //
+        // Constructors
+        //
+        //---------------------------------------------------
+        #region Constructors
+
+        /// <summary>
+        /// In an inherited class, initializes a new instance of 
+        /// the System.Windows.CommonItemDialog class.
+        /// </summary>
+        private protected CommonItemDialog()
+        {
+            // Call Initialize to set defaults for fields
+            // and to set defaults for some option flags.
+            // Initialize() is also called from the virtual
+            // Reset() function to restore defaults.
+            Initialize();
+        }
+
+        #endregion Constructors
+
+        //---------------------------------------------------
+        //
+        // Public Methods
+        //
+        //---------------------------------------------------
+        #region Public Methods
+
+        /// <summary>
+        ///  Resets all properties to their default values.
+        ///  Classes derived from FileDialog are expected to 
+        ///  call Base.Reset() at the beginning of their
+        ///  implementation of Reset() if they choose to
+        ///  override this function.
+        /// </summary>
+        public override void Reset()
+        {
+            Initialize();
+        }
+
+        #endregion Public Methods
+
+        //---------------------------------------------------
+        //
+        // Public Properties
+        //
+        //---------------------------------------------------
+        #region Public Properties
+
+        //   The actual flag is FOS_NODEREFERENCELINKS (set = do not dereference, unset = deref) - 
+        //   while we have true = dereference and false=do not dereference.  Because we expose
+        //   the opposite of the Windows flag as a property to be clearer, we need to negate 
+        //   the value in both the getter and the setter here.
+        /// <summary>
+        ///  Gets or sets a value indicating whether the dialog box returns the location 
```

Comment:
Technically it is correct since it applies to files only. I added a note saying _Not all dialogs allow users to select shortcuts._ Let me know if you were looking for a specific wording. Fixed in https://github.com/dotnet/wpf/pull/7916/commits/543c8098a03629f522d8d86a9666b673a0e0a045